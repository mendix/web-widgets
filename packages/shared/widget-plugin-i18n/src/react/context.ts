import { createContext } from "react";
import { PluginI18n } from "../types";

const contextPath = "com.mendix.widget.plugin.i18n.context" as const;

type I18nContext = React.Context<PluginI18n | null>;

const createI18nContext = (): I18nContext => createContext<PluginI18n | null>(null);

declare global {
    interface Window {
        [contextPath]?: ReturnType<typeof createI18nContext>;
    }
}

export function getGlobalI18nPluginContext(): I18nContext {
    return window[contextPath] ?? createI18nContext();
}
