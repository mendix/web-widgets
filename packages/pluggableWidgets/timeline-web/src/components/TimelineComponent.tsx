import { UngroupedEventsPositionEnum } from "../../typings/TimelineProps";
import { Children, createElement, ReactElement, ReactNode } from "react";
import classNames from "classnames";
import { BasicItemType, CustomItemType, ItemType, ComponentProps } from "../helpers/types";
import { Icon } from "mendix/components/web/Icon";

export default function TimelineComponent(props: ComponentProps): ReactElement {
    return (
        <div className={classNames("widget-timeline-wrapper", props.class)}>
            {getItems(props.data, props.customVisualization, props.groupEvents, props.ungroupedEventsPosition)}
        </div>
    );
}

export function getItems(
    structuredEvents: Map<string, ItemType[]>,
    customVisualization: boolean,
    groupEvents: boolean,
    orphanEventsPlacement: UngroupedEventsPositionEnum
): ReactNode[] {
    const days: ReactNode[] = [];
    const orphanDays: ReactNode[] = [];
    structuredEvents.forEach((eventsOfDay: ItemType[], groupKey: string) => {
        let events;
        let groupHeader: ReactNode = <span>{groupKey}</span>;

        if (!customVisualization) {
            events = getBasicEventsFromDay(eventsOfDay as BasicItemType[]);
        } else {
            events = getCustomEventsFromDay(eventsOfDay as CustomItemType[]);
            const firstEvent = eventsOfDay[0] as CustomItemType;
            if (firstEvent?.groupHeader) {
                groupHeader = firstEvent.groupHeader;
            }
        }

        const constructedDiv = (
            <div className="timeline-date" key={groupKey}>
                {groupEvents && groupKey && <div className="widget-timeline-date-header">{groupHeader}</div>}
                <div
                    className={classNames(
                        "widget-timeline-events-wrapper",
                        !groupEvents || !groupKey ? "widget-timeline-no-divider" : undefined
                    )}
                >
                    <ul>{events}</ul>
                </div>
            </div>
        );
        if (!groupKey) {
            orphanDays.push(constructedDiv);
        } else {
            days.push(constructedDiv);
        }
    });
    if (orphanEventsPlacement === "beginning") {
        days.unshift(orphanDays);
    } else {
        days.push(orphanDays);
    }
    return days;
}

function getBasicEventsFromDay(eventsOfDay: BasicItemType[]): ReactNode[] {
    return eventsOfDay.map((event, index) => (
        <li
            key={index}
            onClick={() => event.action?.execute()}
            className={classNames("widget-timeline-event", event.action ? "clickable" : undefined)}
        >
            <div className="widget-timeline-icon-wrapper">
                {event.icon ? <Icon icon={event.icon} /> : <div className="widget-timeline-icon-circle" />}
            </div>
            <div className="widget-timeline-flex-container widget-timeline-content-wrapper">
                {event.eventDateTime && (
                    <div className="widget-timeline-date-time-wrapper">
                        <p className="widget-eventTime">{event.eventDateTime}</p>
                    </div>
                )}
                <div className="widget-timeline-flex-container widget-timeline-info-wrapper">
                    {event.title && <p className="widget-timeline-title">{event.title}</p>}
                    {event.description && <p className="widget-timeline-description">{event.description}</p>}
                </div>
            </div>
        </li>
    ));
}

function getCustomEventsFromDay(eventsOfDay: CustomItemType[]): JSX.Element[] {
    return eventsOfDay.map((event, index) => (
        <li
            key={index}
            onClick={() => event.action?.execute()}
            className={classNames("widget-timeline-event", event.action ? "clickable" : undefined)}
        >
            <div className="widget-timeline-icon-wrapper">
                {hasChildren(event.icon) ? event.icon : <div className="widget-timeline-icon-circle" />}
            </div>
            <div className="widget-timeline-flex-container widget-timeline-content-wrapper">
                {hasChildren(event.eventDateTime) && (
                    <div className="widget-timeline-date-time-wrapper">{event.eventDateTime}</div>
                )}
                <div className="widget-timeline-flex-container widget-timeline-info-wrapper">
                    {event.title}
                    {event.description}
                </div>
            </div>
        </li>
    ));
}

function hasChildren(element: any): boolean {
    return Children.count((element as ReactElement)?.props.children) > 0;
}
