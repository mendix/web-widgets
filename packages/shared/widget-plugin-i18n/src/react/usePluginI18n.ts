import { useState } from "react";
import { I18nController } from "../controllers/I18nController";
import { PluginI18n } from "../types";

export function usePluginI18n(languageCode: string): PluginI18n | null {
    const [plugin, setPlugin] = useState<PluginI18n | null>(() => {
        new I18nController({
            languageCode,
            onPluginChange: plugin => setPlugin(plugin)
        });

        return null;
    });

    return plugin;
}
