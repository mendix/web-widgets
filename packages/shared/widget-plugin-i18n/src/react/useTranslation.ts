import { useContext } from "react";
import { PluginI18n } from "../types";
import { getGlobalI18nPluginContext } from "./context";

export function useTranslation(): PluginI18n | null {
    return useContext(getGlobalI18nPluginContext());
}
