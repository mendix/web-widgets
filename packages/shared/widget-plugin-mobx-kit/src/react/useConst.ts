import { useRef } from "react";

export function useConst<T>(fn: T): T extends (...args: never) => infer R ? R : T {
    return (useRef<T | null>(null).current ??= typeof fn === "function" ? fn() : fn);
}
