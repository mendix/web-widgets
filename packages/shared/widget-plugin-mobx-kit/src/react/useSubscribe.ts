import { IReactionOptions, reaction } from "mobx";
import { useEffect, useState } from "react";

export function useSubscribe<Store extends object, T, FireImmediately extends boolean = false>(
    init: Store | (() => Store),
    selector: (store: Store) => T,
    options?: IReactionOptions<T, FireImmediately>
): [T, Store] {
    const [store] = useState(() => (typeof init === "function" ? init() : init));
    const [value, setValue] = useState(() => selector(store));

    function setup(): () => void {
        return reaction(() => selector(store), setValue, options);
    }

    // eslint-disable-next-line react-hooks/exhaustive-deps
    useEffect(setup, []);

    return [value, store];
}
