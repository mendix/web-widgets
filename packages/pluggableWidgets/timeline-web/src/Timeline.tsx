import { createElement, ReactElement } from "react";
import { TimelineContainerProps } from "../typings/TimelineProps";
import "./ui/Timeline.scss";
import TimelineComponent from "./components/TimelineComponent";
import { withBasicItems } from "./hocs/withBasicItems";
import { withCustomItems } from "./hocs/withCustomItems";

const BasicView = withBasicItems(TimelineComponent);
const CustomView = withCustomItems(TimelineComponent);

export default function Timeline(props: TimelineContainerProps): ReactElement {
    if (props.customVisualization) {
        return <CustomView {...props} />;
    }

    return <BasicView {...props} />;
}
