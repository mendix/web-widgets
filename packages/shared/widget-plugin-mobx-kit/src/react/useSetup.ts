import { useEffect } from "react";
import { useConst } from "./useConst";

export function useSetup<
    T extends {
        setup(): void | (() => void);
    }
>(fn: () => T): T {
    const obj = useConst(fn);
    useEffect(() => obj.setup(), [obj]);
    return obj;
}
