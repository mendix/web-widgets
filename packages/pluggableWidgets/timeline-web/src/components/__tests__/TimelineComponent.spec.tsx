import { shallow } from "enzyme";
import { createElement } from "react";
import { actionValue, EditableValueBuilder } from "@mendix/widget-plugin-test-utils";
import TimelineComponent, { getGroupHeaderByType, TimelineComponentProps } from "../TimelineComponent";
import { BasicItemType, CustomItemType } from "../../Timeline";

const firstDate = new Date(Date.UTC(1453, 4, 29));
const secondDate = new Date(Date.UTC(1453, 4, 30));

describe("Timeline", () => {
    const basicData = new Map<string, BasicItemType[]>();
    const customData = new Map<string, CustomItemType[]>();

    const basicItem: BasicItemType = {
        description: "Basic description",
        eventDateTime: "Basic event Time",
        title: "Basic title"
    };

    const basicItemWithIcon: BasicItemType = {
        ...basicItem,
        icon: { type: "image", iconUrl: "iconUrl" }
    };

    const customItem: CustomItemType = {
        groupHeader: <p>Day Divider</p>,
        title: <p>Title</p>,
        eventDateTime: <p>Date Time</p>,
        description: <p>Description</p>
    };

    const customItemWithIcon: CustomItemType = {
        ...customItem,
        icon: <img src="customIcon" />
    };

    basicData.set(firstDate.toDateString(), [basicItem, basicItemWithIcon]);
    customData.set(secondDate.toDateString(), [customItem, customItemWithIcon]);

    const basicRenderProps: TimelineComponentProps = {
        data: basicData,
        customVisualization: false,
        groupEvents: true,
        ungroupedEventsPosition: "end"
    };
    const customRenderProps: TimelineComponentProps = {
        data: customData,
        customVisualization: true,
        groupEvents: true,
        ungroupedEventsPosition: "end"
    };

    it("renders timeline with basic configuration", () => {
        const component = shallow(<TimelineComponent {...basicRenderProps} />);
        expect(component).toMatchSnapshot();
    });
    it("renders timeline with custom configuration ", () => {
        const component = shallow(<TimelineComponent {...customRenderProps} />);
        expect(component).toMatchSnapshot();
    });
    it("hides the timeline header", () => {
        const component = shallow(<TimelineComponent {...basicRenderProps} groupEvents={false} />);
        expect(component).toMatchSnapshot();
    });

    it("calls correct formatter with fulldate", () => {
        const date = new EditableValueBuilder<Date>().withValue(firstDate).build();
        getGroupHeaderByType(date.formatter, "fullDate", secondDate);

        expect((date.formatter as any).withConfig).toBeCalledWith({ type: "date" });
    });

    it("calls correct formatter with day", () => {
        const date = new EditableValueBuilder<Date>().withValue(firstDate).build();
        getGroupHeaderByType(date.formatter, "day", secondDate);

        expect((date.formatter as any).withConfig).toBeCalledWith({ type: "date" });
    });

    it("calls correct formatter with dayName", () => {
        const date = new EditableValueBuilder<Date>().withValue(firstDate).build();
        getGroupHeaderByType(date.formatter, "dayName", secondDate);

        expect((date.formatter as any).withConfig).toBeCalledWith({ type: "custom", pattern: "EEEE" });
    });

    it("calls correct formatter with dayMonth", () => {
        const date = new EditableValueBuilder<Date>().withValue(firstDate).build();
        getGroupHeaderByType(date.formatter, "dayMonth", secondDate);

        expect((date.formatter as any).withConfig).toBeCalledWith({ type: "custom", pattern: "dd MMMM" });
    });

    it("calls correct formatter with month", () => {
        const date = new EditableValueBuilder<Date>().withValue(firstDate).build();
        getGroupHeaderByType(date.formatter, "month", secondDate);

        expect((date.formatter as any).withConfig).toBeCalledWith({ type: "custom", pattern: "MMMM" });
    });

    it("calls correct formatter with monthYear", () => {
        const date = new EditableValueBuilder<Date>().withValue(firstDate).build();
        getGroupHeaderByType(date.formatter, "monthYear", secondDate);

        expect((date.formatter as any).withConfig).toBeCalledWith({ type: "custom", pattern: "MMM YYYY" });
    });

    it("calls correct formatter with year", () => {
        const date = new EditableValueBuilder<Date>().withValue(firstDate).build();
        getGroupHeaderByType(date.formatter, "year", secondDate);

        expect((date.formatter as any).withConfig).toBeCalledWith({ type: "custom", pattern: "YYYY" });
    });

    describe("with action set", () => {
        it("renders with clickable styles", () => {
            const action = actionValue(true, false);
            const basicItemWithAction: BasicItemType = {
                ...basicItem,
                icon: { type: "image", iconUrl: "iconUrl" },
                action
            };
            basicData.set(secondDate.toDateString(), [basicItemWithAction]);

            const basicPropsWithAction = { ...basicRenderProps, data: basicData };
            const component = shallow(<TimelineComponent {...basicPropsWithAction} />);

            const clickableItem = component.find(".clickable");

            expect(clickableItem).toHaveLength(1);
            expect(component).toMatchSnapshot();
        });
        it("triggers actions when clicked", () => {
            const action = actionValue(true, false);
            const basicItemWithAction: BasicItemType = {
                ...basicItem,
                icon: { type: "image", iconUrl: "iconUrl" },
                action
            };
            basicData.set(secondDate.toDateString(), [basicItemWithAction]);

            const basicPropsWithAction = { ...basicRenderProps, data: basicData };
            const component = shallow(<TimelineComponent {...basicPropsWithAction} />);

            const clickableItem = component.find(".clickable");
            clickableItem.simulate("click");

            expect(action.execute).toBeCalled();
        });
        it("change style when hovered", () => {
            const action = actionValue(true, false);
            const basicItemWithAction: BasicItemType = {
                ...basicItem,
                icon: { type: "image", iconUrl: "iconUrl" },
                action
            };
            basicData.set(secondDate.toDateString(), [basicItemWithAction]);

            const basicPropsWithAction = { ...basicRenderProps, data: basicData };
            const component = shallow(<TimelineComponent {...basicPropsWithAction} />);

            const clickableItem = component.find(".clickable");
            clickableItem.simulate("mouseover");

            expect(component).toMatchSnapshot();
        });
    });
});
