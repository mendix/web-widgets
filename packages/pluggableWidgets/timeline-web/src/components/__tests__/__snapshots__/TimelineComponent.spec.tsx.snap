// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline hides the timeline header 1`] = `
<div>
  <div
    class="widget-timeline-wrapper"
  >
    <div
      class="timeline-date"
    >
      <div
        class="widget-timeline-events-wrapper widget-timeline-no-divider"
      >
        <ul>
          <li
            class="widget-timeline-event"
          >
            <div
              class="widget-timeline-icon-wrapper"
            >
              <div
                class="widget-timeline-icon-circle"
              />
            </div>
            <div
              class="widget-timeline-flex-container widget-timeline-content-wrapper"
            >
              <div
                class="widget-timeline-date-time-wrapper"
              >
                <p
                  class="widget-eventTime"
                >
                  Basic event Time
                </p>
              </div>
              <div
                class="widget-timeline-flex-container widget-timeline-info-wrapper"
              >
                <p
                  class="widget-timeline-title"
                >
                  Basic title
                </p>
                <p
                  class="widget-timeline-description"
                >
                  Basic description
                </p>
              </div>
            </div>
          </li>
          <li
            class="widget-timeline-event"
          >
            <div
              class="widget-timeline-icon-wrapper"
            >
              <img
                src="mocked/web/icon"
              />
            </div>
            <div
              class="widget-timeline-flex-container widget-timeline-content-wrapper"
            >
              <div
                class="widget-timeline-date-time-wrapper"
              >
                <p
                  class="widget-eventTime"
                >
                  Basic event Time
                </p>
              </div>
              <div
                class="widget-timeline-flex-container widget-timeline-info-wrapper"
              >
                <p
                  class="widget-timeline-title"
                >
                  Basic title
                </p>
                <p
                  class="widget-timeline-description"
                >
                  Basic description
                </p>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Timeline renders timeline with basic configuration 1`] = `
<div>
  <div
    class="widget-timeline-wrapper"
  >
    <div
      class="timeline-date"
    >
      <div
        class="widget-timeline-date-header"
      >
        <span>
          Sun May 29 1453
        </span>
      </div>
      <div
        class="widget-timeline-events-wrapper"
      >
        <ul>
          <li
            class="widget-timeline-event"
          >
            <div
              class="widget-timeline-icon-wrapper"
            >
              <div
                class="widget-timeline-icon-circle"
              />
            </div>
            <div
              class="widget-timeline-flex-container widget-timeline-content-wrapper"
            >
              <div
                class="widget-timeline-date-time-wrapper"
              >
                <p
                  class="widget-eventTime"
                >
                  Basic event Time
                </p>
              </div>
              <div
                class="widget-timeline-flex-container widget-timeline-info-wrapper"
              >
                <p
                  class="widget-timeline-title"
                >
                  Basic title
                </p>
                <p
                  class="widget-timeline-description"
                >
                  Basic description
                </p>
              </div>
            </div>
          </li>
          <li
            class="widget-timeline-event"
          >
            <div
              class="widget-timeline-icon-wrapper"
            >
              <img
                src="mocked/web/icon"
              />
            </div>
            <div
              class="widget-timeline-flex-container widget-timeline-content-wrapper"
            >
              <div
                class="widget-timeline-date-time-wrapper"
              >
                <p
                  class="widget-eventTime"
                >
                  Basic event Time
                </p>
              </div>
              <div
                class="widget-timeline-flex-container widget-timeline-info-wrapper"
              >
                <p
                  class="widget-timeline-title"
                >
                  Basic title
                </p>
                <p
                  class="widget-timeline-description"
                >
                  Basic description
                </p>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Timeline renders timeline with custom configuration 1`] = `
<div>
  <div
    class="widget-timeline-wrapper"
  >
    <div
      class="timeline-date"
    >
      <div
        class="widget-timeline-date-header"
      >
        <p>
          Day Divider
        </p>
      </div>
      <div
        class="widget-timeline-events-wrapper"
      >
        <ul>
          <li
            class="widget-timeline-event"
          >
            <div
              class="widget-timeline-icon-wrapper"
            >
              <div
                class="widget-timeline-icon-circle"
              />
            </div>
            <div
              class="widget-timeline-flex-container widget-timeline-content-wrapper"
            >
              <div
                class="widget-timeline-date-time-wrapper"
              >
                <p>
                  Date Time
                </p>
              </div>
              <div
                class="widget-timeline-flex-container widget-timeline-info-wrapper"
              >
                <p>
                  Title
                </p>
                <p>
                  Description
                </p>
              </div>
            </div>
          </li>
          <li
            class="widget-timeline-event"
          >
            <div
              class="widget-timeline-icon-wrapper"
            >
              <div
                class="widget-timeline-icon-circle"
              />
            </div>
            <div
              class="widget-timeline-flex-container widget-timeline-content-wrapper"
            >
              <div
                class="widget-timeline-date-time-wrapper"
              >
                <p>
                  Date Time
                </p>
              </div>
              <div
                class="widget-timeline-flex-container widget-timeline-info-wrapper"
              >
                <p>
                  Title
                </p>
                <p>
                  Description
                </p>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Timeline with action set change style when hovered 1`] = `
<div>
  <div
    class="widget-timeline-wrapper"
  >
    <div
      class="timeline-date"
    >
      <div
        class="widget-timeline-date-header"
      >
        <span>
          Sun May 29 1453
        </span>
      </div>
      <div
        class="widget-timeline-events-wrapper"
      >
        <ul>
          <li
            class="widget-timeline-event"
          >
            <div
              class="widget-timeline-icon-wrapper"
            >
              <div
                class="widget-timeline-icon-circle"
              />
            </div>
            <div
              class="widget-timeline-flex-container widget-timeline-content-wrapper"
            >
              <div
                class="widget-timeline-date-time-wrapper"
              >
                <p
                  class="widget-eventTime"
                >
                  Basic event Time
                </p>
              </div>
              <div
                class="widget-timeline-flex-container widget-timeline-info-wrapper"
              >
                <p
                  class="widget-timeline-title"
                >
                  Basic title
                </p>
                <p
                  class="widget-timeline-description"
                >
                  Basic description
                </p>
              </div>
            </div>
          </li>
          <li
            class="widget-timeline-event"
          >
            <div
              class="widget-timeline-icon-wrapper"
            >
              <img
                src="mocked/web/icon"
              />
            </div>
            <div
              class="widget-timeline-flex-container widget-timeline-content-wrapper"
            >
              <div
                class="widget-timeline-date-time-wrapper"
              >
                <p
                  class="widget-eventTime"
                >
                  Basic event Time
                </p>
              </div>
              <div
                class="widget-timeline-flex-container widget-timeline-info-wrapper"
              >
                <p
                  class="widget-timeline-title"
                >
                  Basic title
                </p>
                <p
                  class="widget-timeline-description"
                >
                  Basic description
                </p>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="timeline-date"
    >
      <div
        class="widget-timeline-date-header"
      >
        <span>
          Mon May 30 1453
        </span>
      </div>
      <div
        class="widget-timeline-events-wrapper"
      >
        <ul>
          <li
            class="widget-timeline-event clickable"
          >
            <div
              class="widget-timeline-icon-wrapper"
            >
              <img
                src="mocked/web/icon"
              />
            </div>
            <div
              class="widget-timeline-flex-container widget-timeline-content-wrapper"
            >
              <div
                class="widget-timeline-date-time-wrapper"
              >
                <p
                  class="widget-eventTime"
                >
                  Basic event Time
                </p>
              </div>
              <div
                class="widget-timeline-flex-container widget-timeline-info-wrapper"
              >
                <p
                  class="widget-timeline-title"
                >
                  Basic title
                </p>
                <p
                  class="widget-timeline-description"
                >
                  Basic description
                </p>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Timeline with action set renders with clickable styles 1`] = `
<div>
  <div
    class="widget-timeline-wrapper"
  >
    <div
      class="timeline-date"
    >
      <div
        class="widget-timeline-date-header"
      >
        <span>
          Sun May 29 1453
        </span>
      </div>
      <div
        class="widget-timeline-events-wrapper"
      >
        <ul>
          <li
            class="widget-timeline-event"
          >
            <div
              class="widget-timeline-icon-wrapper"
            >
              <div
                class="widget-timeline-icon-circle"
              />
            </div>
            <div
              class="widget-timeline-flex-container widget-timeline-content-wrapper"
            >
              <div
                class="widget-timeline-date-time-wrapper"
              >
                <p
                  class="widget-eventTime"
                >
                  Basic event Time
                </p>
              </div>
              <div
                class="widget-timeline-flex-container widget-timeline-info-wrapper"
              >
                <p
                  class="widget-timeline-title"
                >
                  Basic title
                </p>
                <p
                  class="widget-timeline-description"
                >
                  Basic description
                </p>
              </div>
            </div>
          </li>
          <li
            class="widget-timeline-event"
          >
            <div
              class="widget-timeline-icon-wrapper"
            >
              <img
                src="mocked/web/icon"
              />
            </div>
            <div
              class="widget-timeline-flex-container widget-timeline-content-wrapper"
            >
              <div
                class="widget-timeline-date-time-wrapper"
              >
                <p
                  class="widget-eventTime"
                >
                  Basic event Time
                </p>
              </div>
              <div
                class="widget-timeline-flex-container widget-timeline-info-wrapper"
              >
                <p
                  class="widget-timeline-title"
                >
                  Basic title
                </p>
                <p
                  class="widget-timeline-description"
                >
                  Basic description
                </p>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="timeline-date"
    >
      <div
        class="widget-timeline-date-header"
      >
        <span>
          Mon May 30 1453
        </span>
      </div>
      <div
        class="widget-timeline-events-wrapper"
      >
        <ul>
          <li
            class="widget-timeline-event clickable"
          >
            <div
              class="widget-timeline-icon-wrapper"
            >
              <img
                src="mocked/web/icon"
              />
            </div>
            <div
              class="widget-timeline-flex-container widget-timeline-content-wrapper"
            >
              <div
                class="widget-timeline-date-time-wrapper"
              >
                <p
                  class="widget-eventTime"
                >
                  Basic event Time
                </p>
              </div>
              <div
                class="widget-timeline-flex-container widget-timeline-info-wrapper"
              >
                <p
                  class="widget-timeline-title"
                >
                  Basic title
                </p>
                <p
                  class="widget-timeline-description"
                >
                  Basic description
                </p>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
